src/Catalyst.Core.Modules.Dfs/IpfsAdapter.cs

```diff
@@ -0,0 +1,191 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Catalyst.Abstractions.Cryptography;
+using Catalyst.Abstractions.Dfs;
+using Catalyst.Abstractions.FileSystem;
+using Catalyst.Abstractions.Types;
+using Catalyst.Core.Lib.Config;
+using Common.Logging;
+using Common.Logging.Serilog;
+using LibP2P;
+using LibP2P.Cryptography;
+using Serilog;
+using TheDotNetLeague.Ipfs.Abstractions;
+using TheDotNetLeague.Ipfs.Core.Lib;
+using TheDotNetLeague.MultiFormats.MultiAddress;
+using TheDotNetLeague.MultiFormats.MultiBase;
+
+namespace Catalyst.Core.Modules.Dfs
+{
+    /// <summary>
+    ///   Modifies the IPFS behaviour to meet the Catalyst requirements.
+    /// </summary>
+    /// <remarks>
+    ///   The IPFS engine is lazy, it is only started when needed.
+    /// </remarks>
+    public sealed class IpfsAdapter : IIpfsAdapter
+    {
+        /// <summary>
+        ///   An IPFS implementation, commonly called an IPFS node/
+        /// </summary>
+        private IpfsEngine _ipfs;
+
+        private bool _isStarted;
+        private readonly object _startingLock = new object();
+        private readonly ILogger _logger;
+
+        static IpfsAdapter()
+        {
+            LogManager.Adapter = new SerilogFactoryAdapter(Log.Logger);
+        }
+
+        public IpfsAdapter(IPasswordManager passwordReader,
+            IFileSystem fileSystem,
+            ILogger logger,
+            string swarmKey = "07a8e9d0c43400927ab274b7fa443596b71e609bacae47bd958e5cd9f59d6ca3",
+            IEnumerable<MultiAddress> seedServers = null)
+        {
+            if (seedServers == null || seedServers.Count() == 0)
+            {
+                seedServers = new[]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Introduce a new variable instead of reusing the parameter 'seedServers'.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4318201)